src/app/components/shared/datetime/datetime/datetime.component.ts

```diff
@@ -0,0 +1,132 @@
+import { Component, Input } from "@angular/core";
+import { isInstantiated } from "@helpers/isInstantiated/isInstantiated";
+import { TimezoneInformation } from "@interfaces/apiInterfaces";
+import { DateTime, IANAZone, Zone } from "luxon";
+import { AbstractDatetimeComponent } from "../abstract-datetime.component";
+
+type BawTimezoneUnion = Zone | TimezoneInformation | string;
+
+/**
+ * Displays a datetime (possibly from another timezone) in the users locale and timezone
+ * Commonly used for displaying client side timestamps e.g. Project created at: xyz
+ *
+ * This component does not assume that all dates input are utc, and will localize using ISO8601 offset information if possible
+ */
+@Component({
+  selector: "baw-datetime",
+  templateUrl: "../abstract-datetime.component.html",
+})
+export class DatetimeComponent extends AbstractDatetimeComponent {
+  public constructor() {
+    super();
+  }
+
+  @Input({ required: true }) public value!: DateTime | Date;
+  @Input() public dateOnly: string | boolean;
+  @Input() public timeOnly: string | boolean;
+  protected timezone?: BawTimezoneUnion;
+
+  // we create a getter for the a luxon DateTime object so that we can get values from the @Input() value
+  // without type guards, shape, and instance checks
+  protected get luxonDateTime(): DateTime {
+    const luxonDate: DateTime =
+      this.value instanceof Date
+        ? DateTime.fromJSDate(this.value, { zone: "utc" })
+        : this.value;
+
+    // localize the date/time to the users timezone using toLocal()
+    return luxonDate.toLocal();
+  }
+
```

Comment:
wouldn't this be better modelled as a setter on `value`? If it is read more than once we're doing unnecessary work

Reply:
We can't use a setter since we're using a props `@Input()`, however, we can apply a transform function to the value (similar to the functionality of a setter)

![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/f8661420-72e6-4479-98a4-3bd4dd90b9fd)

In my mind: This satisfies the objective of not having to reshape the object every time we use the value in the component

Reply:
discussed out of band: transform function is for text/template set/attribute values only and not appropriate here

Reply:
Done